subsys/bluetooth/mesh/adv.h

```diff
@@ -25,40 +25,49 @@ enum bt_mesh_adv_type {
 	BT_MESH_ADV_TYPES,
 };
 
-typedef void (*bt_mesh_adv_func_t)(struct net_buf *buf, uint16_t duration,
-				   int err, void *user_data);
+enum bt_mesh_adv_tag {
+	BT_MESH_LOCAL_ADV,
+	BT_MESH_RELAY_ADV,
+
+	/* Only used with bitfield */
+	BT_MESH_PROXY_ADV,
+};
+
+/* values for tag bitfield */
+#define BT_MESH_ADV_TAG_LOCAL	BIT(BT_MESH_LOCAL_ADV)
+#define BT_MESH_ADV_TAG_RELAY	BIT(BT_MESH_RELAY_ADV)
+#define BT_MESH_ADV_TAG_PROXY	BIT(BT_MESH_PROXY_ADV)
 
 struct bt_mesh_adv {
 	const struct bt_mesh_send_cb *cb;
 	void *cb_data;
 
 	uint8_t      type:2,
 		  started:1,
-		  busy:1;
+		  busy:1,
+		  tag:1;
```

Comment:
This cannot work. You’re reserving a single bit for a value that needs at least three bits.

Reply:
``` C
enum bt_mesh_adv_tag {
	BT_MESH_LOCAL_ADV,
	BT_MESH_RELAY_ADV,

	/* Only used with bitfield */
	BT_MESH_PROXY_ADV,
};

```

Actually, we don't use `BT_MESH_PROXY_ADV` when call `bt_mesh_adv_buf_send`();

Reply:
In order to avoid misunderstanding, I specially added a note `	/* Only used with bitfield */
 `.

Reply:
What got me confused is that you're defining two sets of values for the different types of packets. One set is made up of mutually orthogonal values (using the `BIT()` macro) that can be logically or-ed together, and another is an enum. I thought you were assigning the `BIT()` values to the one bit wide bitfield.

Reply:
What would help, I think, is to have a comment for this bit field that only the enum values are assigned to it, and not the `BT_MESH_ADV_TAG_*` values. The comment you're referring to in the enum definition didn't really help in anyway to clarify this detail.

Reply:
If `BIT() `definitions are used, it will occupy two extra bits. I just hope to leave more space for the future` uint8_t      type:2,`

Maybe I need to ask for @trond-snekvik  advice.

Reply:
Alternatively, since there is space for it, extend the width of the field to three bits and completely remove the confusing duplicate enumeration, i.e. only use the `BIT()` values.

Reply:
Done.